app/repositories/submission_repository.rb

```diff
@@ -11,14 +11,14 @@ def to_rate
     to_rate_scope.to_a
   end
 
-  def next_to_rate(current_created_at)
-    to_rate_scope.where('submissions.created_at > ?', current_created_at).order('created_at ASC')
-      .first || to_rate_scope.first
+  def next(current_created_at)
+    not_rejected.where('submissions.created_at > ?', current_created_at).order('created_at ASC')
+      .first || not_rejected.first
```

Comment:
this `||` is here for not passing `current_created_at` case? maybe it would be better to be more specific and have `if` using `current_created_at `?

Something like that:

```ruby
scope = if current_created_at
  not_rejected.where('submissions.created_at > ?', current_created_at)
else
  not_rejected
end
scope.order('created_at ASC').first
```

Also probably you need ordering in both cases?